File: api/get-audio.py
import yt_dlp
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/get-audio', methods=['GET'])
def get_audio():
    video_url = request.args.get('url')
    if not video_url:
        return jsonify({"error": "YouTube URL is required."}), 400

    try:
        # Use yt-dlp to extract audio URL
        ydl_opts = {'format': 'bestaudio'}
        with yt_dlp.YoutubeDL(ydl_opts) as ydl:
            info = ydl.extract_info(video_url, download=False)
            audio_url = info['url']
            return jsonify({"audioUrl": audio_url}), 200
    except Exception as e:
        return jsonify({"error": str(e)}), 500

# Vercel requires this handler for serverless functions
def handler(event, context):
    from flask_lambda import FlaskLambda
    app_lambda = FlaskLambda(app)
    return app_lambda(event, context)

File: api/get-audio.js
const { exec } = require("child_process");
const path = require("path");

// Function to extract audio URL
export default function handler(req, res) {
  if (req.method !== "GET") {
    return res.status(405).json({ error: "Method not allowed. Use GET." });
  }

  const videoUrl = req.query.url;

  if (!videoUrl) {
    return res.status(400).json({ error: "YouTube URL is required." });
  }

  const command = `yt-dlp -g -f bestaudio ${videoUrl}`;
  exec(command, { cwd: path.resolve() }, (error, stdout, stderr) => {
    if (error) {
      return res.status(500).json({ error: `Error: ${stderr || error.message}` });
    }

    const audioUrl = stdout.trim();
    res.status(200).json({ audioUrl });
  });
}
